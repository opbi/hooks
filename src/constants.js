export const FOO = 'hello';
export const BAR = 'world';
