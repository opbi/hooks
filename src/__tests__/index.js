import example from 'index';

import * as constants from 'constants';

jest.mock('constants');

describe('example', () => {
  it('works', () => {
    const result = example();
    expect(result).toBe('hello world');
  });

  it('would fail if constant types corrupt', () => {
    constants.FOO = 1;
    constants.BAR = 'foo';
    expect(example).toThrowErrorMatchingSnapshot();
  });
});
