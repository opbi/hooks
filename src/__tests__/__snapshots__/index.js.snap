// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`example would fail if constant types corrupt 1`] = `"function validation failed"`;
